compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -298,6 +310,54 @@ BLangStatementExpression desugar(BLangQueryAction queryAction, SymbolEnv env) {
         return stmtExpr;
     }
 
+    /**
+     * Get result type of the query output.
+     *
+     * @param type type of query expression.
+     * @return result type.
+     */
+    private BType resolveExprType(BType type) {
+        if (type.tag == TypeTags.STREAM) {
+            return ((BStreamType) type).constraint;
+        } else if (type.tag == TypeTags.TABLE) {
+            return ((BTableType) type).constraint;
+        } else if (type.tag == TypeTags.ARRAY) {
+            return ((BArrayType) type).eType;
+        } else if (type.tag == TypeTags.UNION) {
+            List<BType> exprTypes = new ArrayList<>(((BUnionType) type).getMemberTypes());
+            for (BType t : exprTypes) {
+                BType returnType;
+                if (t.tag == TypeTags.STREAM) {
+                    returnType = ((BUnionType) type).getMemberTypes()
+                            .stream().filter(m -> m.tag == TypeTags.STREAM)
+                            .findFirst().orElse(symTable.streamType);
+                    return ((BStreamType) returnType).constraint;
+                } else if (t.tag == TypeTags.TABLE) {
+                    returnType = ((BUnionType) type).getMemberTypes()
```

Comment:
why are we filtering again? can't we have just the "t"? When can orElse be executed?